metric_tank/cassandra.go

```diff
@@ -71,20 +71,32 @@ func InitCassandra() error {
 	cluster.NumConns = *cassandraWriteConcurrency
 	cSession, err = cluster.CreateSession()
 
-	CassandraWriteQueue = make(chan *ChunkWriteRequest, *cassandraWriteQueueSize)
+	CassandraWriteQueue = make(chan *ChunkWriteRequest)
+	workerQueues := make([]chan *ChunkWriteRequest, *cassandraWriteConcurrency)
 	for i := 0; i < *cassandraWriteConcurrency; i++ {
-		go processWriteQueue()
+		workerQueues[i] = make(chan *ChunkWriteRequest, *cassandraWriteQueueSize)
+		go processWriteQueue(workerQueues[i])
 	}
+	go func() {
```

Comment:
cassandraWriteQueue is unbuffered and this is a single goroutine consuming it. if this goroutine cant keep up with the rate of messages it will block the caller which will lead to NSQ messages not being processed.